### PR TITLE
RUN-2913 adds ability to launch apps from manifest programmatically

### PR DESCRIPTION
**Pull request by whyn07m3**
ℹ️  This PR adds ability to launch app from a manifest programmatically. It supports multi-runtime app creation (you get a success callback once the app in another runtime launches).

✅  [Tests okay](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5910d6c45114ac10ca6db370)

👫  [Coupled with this PR in javascript-adapter](https://github.com/openfin/javascript-adapter/pull/304)

➕  New tests (will publish at the right time):
1. [Application.createFromManifest](https://testing-dashboard.openfin.co/#/app/tests/5911cbc55114ac10ca6db372/edit)
2. [Application.createFromManifest T2](https://testing-dashboard.openfin.co/#/app/tests/5910e1105114ac10ca6db371/edit)
3. [Application.run (for Application.createFromManifest)](https://testing-dashboard.openfin.co/#/app/tests/5911ccbe5114ac10ca6db373/edit)
4. [Application.run T2 (for Application.createFromManifest)](https://testing-dashboard.openfin.co/#/app/tests/5911da355114ac10ca6db375/edit)
5. [Application.run (multi-runtime for Application.createFromManifest)](https://testing-dashboard.openfin.co/#/app/tests/5911dba25114ac10ca6db376/edit)
6. [Application.run T2 (multi-runtime for Application.createFromManifest)](https://testing-dashboard.openfin.co/#/app/tests/5911dc045114ac10ca6db377/edit)

### CODE REVIEW

**Comment by whyn07m3** [comment]
✅  [Tests results look good](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/591618ad5114ac10ca6db3a9)

**Comment by whyn07m3** [comment]
✅  [Test results](https://testing-dashboard.openfin.co/#/app/sessions)

File: src/browser/core_state.js [hunk 1]
```diff
@@ -499,6 +499,19 @@ function getSocketServerState() {
     return socketServerState;
 }
 
+/**
+ * Get app's very first ancestor
+ */
+function getAppAncestor(descendantAppUuid) {
+    const app = appByUuid(descendantAppUuid);
+
+    if (app && app.appObj && app.appObj.parentUuid) {
```
Comment (HarsimranSingh) [inline src/browser/core_state.js]: I think @datamadic changed this so the `parentUuid` is now on the `app` level, not on the `appObj` level.

Comment (HarsimranSingh) [inline src/browser/core_state.js]: This commit https://github.com/HadoukenIO/core/commit/a91bcbcd6957336594f95f9cfcda80ce4328a1ef

Comment (datamadic) [inline src/browser/core_state.js]: 👍 nice catch @HarsimranSingh

Comment (whyn07m3) [inline src/browser/core_state.js]: Woah! Nice! 🥇

**Comment by HarsimranSingh** [comment]
👍

**Pull request merged**